cmd/docker-app/merge.go

```diff
@@ -32,13 +69,25 @@ func mergeCmd(dockerCli command.Cli) *cobra.Command {
 				if err != nil {
 					return err
 				}
-				defer target.(io.WriteCloser).Close()
 			}
-			return packager.Merge(appname, target)
+			if err := packager.Merge(extractedApp.AppName, target); err != nil {
+				return err
+			}
+			if mergeOutputFile != "-" {
+				// Need to close for the Rename to work on windows.
+				target.(io.WriteCloser).Close()
+			}
+			if inPlace {
+				if err := os.RemoveAll(extractedApp.OriginalAppName); err != nil {
+					return errors.Wrap(err, "failed to erase previous application")
+				}
+				if err := os.Rename(mergeOutputFile, strings.TrimSuffix(mergeOutputFile, ".tmp")); err != nil {
```

Comment:
nit: `strings.TrimSuffix(mergeOutputFile, ".tmp")` should be `extractedApp.OriginalAppName` (the intent is clearer)